include/dlaf/matrix/tile.h

```diff
@@ -426,113 +416,67 @@ pika::shared_future<Tile<T, D>> splitTileInsertFutureInChain(pika::future<Tile<T
 /// The next dependency in the dependency chain will become ready only when @p tile
 /// and the returned subtile go out of scope.
 template <class T, Device D>
-pika::shared_future<Tile<const T, D>> splitTile(const pika::shared_future<Tile<const T, D>>& tile,
-                                                const SubTileSpec& spec) {
-  return internal::createSubTile(tile, spec);
+ReadOnlyTileSender<T, D> subTileSender(ReadOnlyTileSender<T, D> tile, const SubTileSpec& spec) {
+  return std::move(tile) |
+         pika::execution::experimental::let_value(
+             [spec](TileAsyncRwMutexReadOnlyWrapper<T, D>& tile_wrapper) {
+               TileAsyncRwMutex<T, D> tile_manager{
+                   internal::createSubTileAsyncRwMutex<T, D>(std::move(tile_wrapper), spec)};
+               return tile_manager.read();
+             }) |
+         pika::execution::experimental::split();
 }
 
-/// Create read-only subtiles of a given read-only tile.
-///
-/// The returned subtiles will get ready, at the same time as @p tile gets ready.
-/// The next dependency in the dependency chain will become ready only when @p tile
-/// and all the returned subtiles go out of scope.
+// TODO: Docs.
+// TODO: shareReadWriteTile?
 template <class T, Device D>
-std::vector<pika::shared_future<Tile<const T, D>>> splitTile(
-    const pika::shared_future<Tile<const T, D>>& tile, const std::vector<SubTileSpec>& specs) {
-  std::vector<pika::shared_future<Tile<const T, D>>> ret;
-  ret.reserve(specs.size());
-  for (const auto& spec : specs) {
-    ret.emplace_back(internal::createSubTile(tile, spec));
-  }
-
-  return ret;
+ReadOnlyTileSender<T, D> shareReadwriteTile(ReadWriteTileSender<T, D>&& tile) {
+  return std::move(tile) | pika::execution::experimental::let_value([](Tile<T, D>& tile) {
+           TileAsyncRwMutex<T, D> tile_manager{std::move(tile)};
+           return tile_manager.read();
+         }) |
+         pika::execution::experimental::split();
```

Comment:
Is `split` necessary? Add a test.

Reply:
It is actually necessary (for now...?) so I'm leaving it. It's not clear to me why though. I might try to look into it separately: https://github.com/pika-org/pika/issues/654.

Reply:
I have added tests for `shareReadWriteTile` though, so I'm closing this as resolved.